src/main/perl/ComponentProxyList.pm

```diff
@@ -58,55 +58,55 @@ sub reportComponents
     return SUCCESS;
 }
 
-# Run the pre-config $hook, possibly timing out after $timeout seconds
+
+# common code for pre/post hooks
+# Run the pre/post-config $hook, possibly timing out after $timeout seconds
+# If data is defined, it is passed via stdin in JSON format
 # Returns 1 on success, 0 on failure
-sub pre_config_actions
+# TODO: the types are probably not preserved all the way here,
+#   so the JSON format e.g. might have strings for integers
+sub _config_actions
```

Comment:
How does this refactoring relate to the Perl warnings during tests? Should this be a separate commit?